src/org/labkey/test/util/exp/SampleTypeAPIHelper.java

```diff
@@ -33,15 +36,72 @@
      */
     public static TestDataGenerator createEmptySampleType(String containerPath, SampleTypeDefinition def)
     {
+        return createEmptySampleType(containerPath, def, null);
+    }
+
+    static public TestDataGenerator createEmptySampleType(String containerPath, SampleTypeDefinition sampleTypeDefinition, String dateFormatString)
+    {
+        deleteDomain(new FieldDefinition.LookupInfo(containerPath, "samples", sampleTypeDefinition.getName()));
+
+        TestDataGenerator dgen;
         try
         {
-            return TestDataGenerator.createDomain(containerPath, def);
+            dgen = TestDataGenerator.createDomain(containerPath, sampleTypeDefinition);
         }
         catch (CommandException e)
         {
             throw new RuntimeException("Failed to create sample type.", e);
         }
 
+        if (null != dateFormatString)
+        {
+            dgen.addDataSupplier("DateColumn", () -> dgen.randomDateString(dateFormatString,
+                    DateUtils.addWeeks(new Date(), -39),
+                    new Date()));
+        }
```

Comment:
It actually sets a data supplier on the TestDataGenerator it returns; as I think on it, the calling code could easily do that just as well.  If there was a way for TestDataGenerator to get the server dateFormatString independently it could do that by itself, but for now it will be something the test will have to do